index.js

```diff
@@ -0,0 +1,133 @@
+const fs = require('fs');
+
+
+class Cookie{
+    constructor(name, ingredients){
+        this._name = name
+        this._ingredients = this.breakdownIngredients(ingredients)
+        this._status = 'mentah'
+    }
+
+    //setter&getter
+    get name(){return this._name}
+    set name(p){this._name=p}
+    get ingredients(){return this._ingredients}
+    set ingredients(p){this._ingredients=p}
+    get status(){return this._status}
+    set status(p){this._status=p}
+
+    breakdownIngredients(ingredients){
```

Comment:
Ingredients dibuat class baru aja. terus nanti di `composition` ke dalam `Cookie` dengan property `ingredients:` (berisi array of object ingredients).

Jadi nanti outputnya seperti ini :
`{
  name: 'red velvet',
  ingredients: [
    Ingredient { name: 'flour', amount: '1 cup' },
    Ingredient { name: 'mild cocoa flavor', amount: '2 cup' },
    Ingredient { name: 'tangy buttermilk', amount: '2 cup' },
    Ingredient { name: 'sweet vanilla', amount: '1 cup' },
    Ingredient { name: 'very buttery \r', amount: '1 cup' }
  ],
  form: 'cake'
}
{
  name: 'chocolate butter',
  ingredients: [
    Ingredient { name: 'glutten free flour', amount: '1 cup' },
    Ingredient { name: 'flavor adders', amount: '1 cup' },
    Ingredient { name: 'butter', amount: '2 tsp' }
  ],
  form: 'cake'
}`